goldenmaster/Plugins/ApiGear/Source/ApiGear/Public/ApiGearConnectionsStore.h

```diff
@@ -0,0 +1,65 @@
+// Copyright Epic Games, Inc. All Rights Reserved
+#pragma once
+
+#include "CoreMinimal.h"
+#include "UObject/ObjectMacros.h"
+#include "UObject/Object.h"
+#include "Subsystems/EngineSubsystem.h"
+#include "Engine/EngineTypes.h"
+#include "Templates/SharedPointer.h"
+#include "ApiGearConnection.h"
+#include "ApiGearConnectionsStore.generated.h"
+
+/**
+ * Implements the connections store for the ApiGear plugin.
+ */
+UCLASS(BlueprintType)
+class APIGEAR_API UApiGearConnectionsStore : public UEngineSubsystem
+{
+	GENERATED_BODY()
+
+public:
+	UApiGearConnectionsStore();
+
+	/** A type of function for creating connections*/
+	using FConnectionFactoryFunction = TFunction<TScriptInterface<IApiGearConnection>(UObject*, FString)>;
+
+	// register factories for different types of connections
+	bool RegisterConnectionFactory(FString ConnectionTypeIdentifier, FConnectionFactoryFunction FactoryFunction);
+
+	// USubssystem functions
+	void Initialize(FSubsystemCollectionBase& Collection) override;
+	void Deinitialize() override;
+
+	UFUNCTION(BlueprintCallable, Category = "ApiGear|Connection")
+	TScriptInterface<IApiGearConnection> GetConnection(FString UniqueEndpointIdentifier) const;
+
+	UFUNCTION(BlueprintCallable, Category = "ApiGear|Connection")
+	TMap<FString, TScriptInterface<IApiGearConnection>> GetConnections() const;
+
+	bool AddConnection(TScriptInterface<IApiGearConnection> Connection);
+	UFUNCTION(BlueprintCallable, Category = "ApiGear|Connection")
+	bool AddConnection(FString UniqueEndpointIdentifier, TScriptInterface<IApiGearConnection> Connection);
+
+	UFUNCTION(BlueprintCallable, Category = "ApiGear|Connection")
+	bool RemoveConnection(FString UniqueEndpointIdentifier);
+
+	UFUNCTION(BlueprintCallable, Category = "ApiGear|Connection")
+	bool GetIsConnected(FString UniqueEndpointIdentifier) const;
```

Comment:
I'd maybe resign from this function, the getConnection should be enough, the user can check its state and settings or connect /disconnect each connections